### PR TITLE
Harvest index page improvement

### PR DESCRIPTION
**Pull request by peterVorman**

### CODE REVIEW

**Comment by jguo144** [comment]
I recommend adding a new template snippet and including it. This way when we need to merge master it will be easier to resolve conflicts. Also the use of `from ckan.logic import NotFound` can be replaced with an import from `toolkit` for future compatibility.

**Comment by peterVorman** [comment]
> I recommend adding a new template snippet and including it. This way when we need to merge master it will be easier to resolve conflicts. Also the use of `from ckan.logic import NotFound` can be replaced with an import from `toolkit` for future compatibility.

removed

File: ckanext/harvest/helpers.py [hunk 1]
```diff
@@ -157,3 +156,16 @@ def bootstrap_version():
...
+
+
+def get_latest_job(harvest_id):
+    context = {'model': model, 'session': model.Session, 'user': p.toolkit.c.user or p.toolkit.c.author}
+    try:
+        hs = p.toolkit.get_action('harvest_source_show')(context, {'id': harvest_id})
+        last_job = hs.get('status', {}).get('last_job', {})
+        if not last_job:
+            return {}
+        job = p.toolkit.get_action('harvest_job_show')(context, {'id': last_job.get('id')})
```
Comment (jguo144) [inline ckanext/harvest/helpers.py]: The `harvest_job_show` action will only work for logged in users. The `harvest_source_show` action will work for everyone.
Add an exception for not authorized.

File: ckanext/harvest/templates/snippets/harvest_source_job_details.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        </td>
+    </tr>
+    <tr>
+        <th>{{ _('Finished') }}</th>
+        <td>
+        <span class="automatic-local-datetime" data-datetime="{{ h.render_datetime(job.finished, date_format='%Y-%m-%dT%H:%M:%S%z') }}">
+            {{ h.render_datetime(job.finished, with_hours=True) }}
+        </span>
+        </td>
+    </tr>
```
Comment (jguo144) [inline ckanext/harvest/templates/snippets/harvest_source_job_details.html]: The rows for job status should be hidden if there are no jobs or if a public user is looking at it.

File: ckanext/harvest/helpers.py (hunk 1 above)
Comment (peterVorman) [reply ckanext/harvest/helpers.py]: done

File: ckanext/harvest/templates/snippets/harvest_source_job_details.html (hunk 2 above)
Comment (peterVorman) [reply ckanext/harvest/templates/snippets/harvest_source_job_details.html]: done

**Review by jguo144 — APPROVED** [review]
LGTM

**Pull request merged**